packages/ndk-multilib/build.sh

```diff
@@ -37,11 +37,51 @@ prepare_libs() {
 	fi
 
 	cp $LIBATOMIC/libatomic.a $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/$SUFFIX/lib/
+
+	cp $LIBATOMIC/libunwind.a $TERMUX_PKG_MASSAGEDIR/$TERMUX_PREFIX/$SUFFIX/lib/
+}
+
+add_cross_compiler_rt() {
+	RT_PREFIX=$NDK/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/*/lib/linux
+	RT_OPT_DIR=$TERMUX_PREFIX/opt/ndk-multilib/cross-compiler-rt
+	mkdir -p $TERMUX_PKG_MASSAGEDIR/$RT_OPT_DIR
+	cp $RT_PREFIX/* $TERMUX_PKG_MASSAGEDIR/$RT_OPT_DIR || true
 }
 
 termux_step_extract_into_massagedir() {
 	prepare_libs "arm" "arm-linux-androideabi"
 	prepare_libs "arm64" "aarch64-linux-android"
 	prepare_libs "x86" "i686-linux-android"
 	prepare_libs "x86_64" "x86_64-linux-android"
+	add_cross_compiler_rt
+}
+
+termux_step_create_debscripts() {
+	LIBLLVM_VERSION=$(. $TERMUX_SCRIPTDIR/packages/libllvm/build.sh; echo $TERMUX_PKG_VERSION)
+	RT_PATH=$TERMUX_PREFIX/lib/clang/$LIBLLVM_VERSION/lib/android
+	# Install the symlinks.
+	cat <<- POSTINST_EOF > ./postinst
+	#!$TERMUX_PREFIX/bin/sh
+	echo "Installing symlinks to $RT_PATH..."
+	mkdir -p $RT_PATH
+	find $RT_OPT_DIR -type f ! -name "libclang_rt*\$(dpkg --print-architecture)*" -exec ln -sfr "{}" $RT_PATH \;
+	exit 0
+	POSTINST_EOF
+
+	# Uninstall the symlinks.
+	cat <<- PRERM_EOF > ./prerm
+	#!$TERMUX_PREFIX/bin/sh
+	if [ "$TERMUX_PACKAGE_FORMAT" != "pacman" ] && [ "\$1" != "remove" ]; then
+	    exit 0
+	fi
+	echo "Uninstalling symlinks..."
+	find $RT_PATH -type l ! -name "libclang_rt*\$(dpkg --print-architecture)*" -exec rm -rf "{}" \;
+	if [ ! "\$(ls -A $RT_PATH)" ]; then
+	    # $RT_PATH is empty. Package libcompiler-rt should have been removed. Remove this folder.
+	    rm -rf $TERMUX_PREFIX/lib/clang
```

Comment:
This smells dangerous to me. Is the removal needed in the first place? If so, then can this be more graceful?

Reply:
Maybe we should create a `prerm` script in `libcompiler-rt` to check and remove these symlinks.

Reply:
Not symlinks but directories. Symlink removal should be handled here.

Reply:
Emmm... In my point of view, this folder should be handled by `dpkg` when `libcompiler-rt` installs.
`prerm` in `libcompiler-rt` and `prerm` in `ndk-multilib` should remove these symlinks if they exist.
`postinst` in `libcompiler-rt` should install these symlinks if `ndk-multilib` is installed.
`postinst` in `ndk-multilib` should install these symlinks if `libcompiler-rt` is installed.
The solution above seems more graceful. Any ideas?

Reply:
Note that there is no dependency between `libcompiler-rt` and `ndk-multilib`.

I'm strongly against

```bash
rm -rf $TERMUX_PREFIX/lib/clang
```

because there is no assurance that this directory does not contain any other files. It would be much more tolerable if this directory were kept unremoved.

Reply:
How about using `-p` option for `rmdir`:

```bash
rmdir -p $RT_PATH || :
```

With this emptyness check can be omitted.

Reply:
Maybe we can touch a random-named file like this?
```
mkdir -p $TERMUX_PKG_MASSAGEDIR/$RT_PATH
touch $TERMUX_PKG_MASSAGEDIR/$RT_PATH/".termux-$(cat /proc/sys/kernel/random/uuid)"
```
Then this folder should be able to create/remove by `dpkg`.

Reply:
No need to be random, unless conflicting with other packages. I wonder if there is any convention, but I would prefer `.keep-ndk-multilib` or something that has package name as suffix.

Reply:
OK. I will fix it.